core/genesis.go

```diff
@@ -183,7 +183,11 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, constant
 			return genesis.Config, hash, &GenesisMismatchError{stored, hash}
 		}
 		block, err := genesis.Commit(db)
-		return genesis.Config, block.Hash(), err
+		var hash common.Hash
+		if block != nil {
+			hash = block.Hash()
+		}
+		return genesis.Config, hash, err
```

Comment:
This variant would be more in line with how we usually do it:
```golang
if block, err := genesis.Commit(db); err != nil{
	return nil, nil, err
}else{
	return genesis.Config, block.Hash(), nil
)
```

Reply:
```go
block, err := genesis.Commit(db)
if err != nil{
	return nil, nil, err
}
return genesis.Config, block.Hash(), nil
```